commerce_coordinator/apps/lms/views.py

```diff
@@ -219,25 +221,49 @@ def post(self, request) -> Response:
+                return Response('Invalid input provided', status=HTTP_400_BAD_REQUEST)
+
+            username = input_details.data['username']
+            order_number = input_details.data.get('order_number')
+            entitlement_id = input_details.data.get('entitlement_id')
+            logger.info(f"[RefundView] Starting LMS Refund of Entitlement for username: {username}, "
+                        f"order_number: {order_number}, Entitlement ID: {entitlement_id}.")
+
+            try:
+                order_id, order_line_item_id = get_line_item_from_entitlement(order_number, entitlement_id)
+
+            except CommercetoolsError as err:  # pragma no cover
+                logger.exception(f"[RefundView] Commercetools Error: {err}, {err.errors}")
+                return Response('Error while fetching order', status=HTTP_400_BAD_REQUEST)
 
-        input_details = CourseRefundInputSerializer(data=input_data)
-        try:
-            input_details.is_valid(raise_exception=True)
-        except ValidationError as e:
-            logger.exception(f"[RefundView] Exception raised validating input {self.post.__name__} with error "
-                             f"{repr(e)}, input: {input_data}.")
-            return Response('Invalid input provided', status=HTTP_400_BAD_REQUEST)
+        else:
+            input_details = CourseRefundInputSerializer(data=input_data)
+            try:
+                input_details.is_valid(raise_exception=True)
+            except ValidationError as e:
+                logger.exception(f"[RefundView] Exception raised validating input {self.post.__name__} with error "
+                                 f"{repr(e)}, input: {input_data}.")
+                return Response('Invalid input provided', status=HTTP_400_BAD_REQUEST)
 
-        course_id = input_details.data['course_id']
-        username = input_details.data['username']
+            course_id = input_details.data['course_id']
+            username = input_details.data['username']
 
-        enrollment_attributes = input_details.enrollment_attributes_dict()
+            enrollment_attributes = input_details.enrollment_attributes_dict()
 
-        logger.info(f"[RefundView] Starting LMS Refund for username: {username}, course_id: {course_id}, "
-                    f"Enrollment attributes: {enrollment_attributes}.")
+            logger.info(f"[RefundView] Starting LMS Refund for username: {username}, course_id: {course_id}, "
+                        f"Enrollment attributes: {enrollment_attributes}.")
 
-        order_line_item_id = enrollment_attributes.get(enrollment_attribute_key('order', 'line_item_id'), None)
-        order_id = enrollment_attributes.get(enrollment_attribute_key('order', 'order_id'), None)
+            order_line_item_id = enrollment_attributes.get(enrollment_attribute_key('order', 'line_item_id'), None)
+            order_id = enrollment_attributes.get(enrollment_attribute_key('order', 'order_id'), None)
```

Comment:
the order_id verification should be inside the else block right? because `if not order_id` cannot be the case for entitlements refund? if there is not order id, it should be handled as part of serializer validation, right?

but if we still want to put `if not order_id` outside of if-else, we should make the message generic.

Reply:
serializer is expecting `order_number` for entitlement so order id won't be verified in serializer. If an order does not found against the given order number, an exception is raised and returned. So in case, order id cannot be empty. Regarding adding it in else, yes we can safely move it to course code